### PR TITLE
Added imposter yaml

### PR DESCRIPTION
**Pull request by surajkumar**
### What changed
Added `public-api-config.yaml` and `private-api-config.yaml`

### Why did it change

So we can run tests using imposter.

### Issue tracking

- [OJ-1398](https://govukverify.atlassian.net/browse/OJ-1398)

### Environment variables or secrets

- [X] No environment variables or secrets were added or changed

[OJ-1398]: https://govukverify.atlassian.net/browse/OJ-1398?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/private-api-config.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+specFile: ./private-api.yaml
+
+validation:
+  request: log
+
+resources:
+  - method: GET
+    path: /authorization
+    requestHeaders:
+      SessionHeader: session-id
```
Comment (scronk1) [inline infrastructure/private-api-config.yaml]: `SessionHeader` is just the term we're using to reference in that spec - it's not something that actually goes into the query. To achieve this we need to do:
```
requestHeaders:
    session-id: <whatever you want to match on>
```

File: infrastructure/private-api-config.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    response:
+      statusCode: 200
+
+  - method: GET
+    path: /authorization
+    requestHeaders:
+      SessionHeader: ""
+    response:
+      statusCode: 400
```
Comment (scronk1) [inline infrastructure/private-api-config.yaml]: What use case are you hoping to achieve here? They have a session-id header but it's blank / the wrong thing?

File: infrastructure/private-api-config.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  - method: GET
+    path: /authorization
+    requestHeaders:
```
Comment (scronk1) [inline infrastructure/private-api-config.yaml:25]: This doesn't need `requestHeaders` - if we want this to be the default behaviour if the 200 and 400 criteria aren't matched then you can just keep it super simple

File: infrastructure/private-api-config.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    response:
+      statusCode: 500
+
+  - method: POST
+    path: /session
+    requestBody:
+      jsonPath: $.client_id
+      value: ipv-core-stub
```
Comment (scronk1) [inline infrastructure/private-api-config.yaml]: Can this be `ipv-toy-stub` or something - core stub is a term used for a different stub so we don't want confusion

File: infrastructure/private-api-config.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    path: /session
+    requestBody:
+      value: ""
+    response:
+      statusCode: 400
+
+  - method: POST
+    path: /session
+    requestBody:
+      value: invalid
```
Comment (scronk1) [inline infrastructure/private-api-config.yaml]: How are you triggering this in a request? I'm trying multiple things and just getting 400

File: infrastructure/private-api-config.yaml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    response:
+      statusCode: 500
+
+  - method: POST
+    path: /favourite
+    requestHeaders:
+      SessionHeader: session-id
```
Comment (scronk1) [inline infrastructure/private-api-config.yaml]: same as above - `SessionHeader` not necessary

File: infrastructure/private-api-config.yaml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    requestBody:
+      jsonPath: $.toy
+      value: football
```
Comment (scronk1) [inline infrastructure/private-api-config.yaml]: Can these examples match the third party stub spec we've got for the toy api - https://github.com/alphagov/di-ipv-third-party-stubs/blob/main/stubs/toy/api-config.yaml

File: infrastructure/public-api-config.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+validation:
+  request: log
+
+resources:
+
+  - method: POST
+    path: /token
+    formParams:
+      code: aaaaaaaaa
```
Comment (scronk1) [inline infrastructure/public-api-config.yaml]: I love the energy but can we have something a bit more relevant?

File: infrastructure/public-api-config.yaml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+  - method: POST
+    path: /credential/issue
+    response:
+      statusCode: 200
+
+  - method: POST
+    path: /credential/issue
+    requestHeaders:
+      SessionHeader: Authorization Bearer 123456789
```
Comment (scronk1) [inline infrastructure/public-api-config.yaml]: Same as above - this should be session-id not SessionHeader
Also I think you've got a duplicate with the block above, shouldn't have more than one 200 response

**Review by scronk1 — CHANGES_REQUESTED** [review]
The API paths are all good 👍  Just some imposter config issues

File: infrastructure/private-api-config.yaml (hunk 5 above)
Comment (surajkumar) [reply infrastructure/private-api-config.yaml]: setting the value to invalid in the body? I can refactor this for easier usage.

**Comment by surajkumar** [comment]
Closing this PR in favour for the new branch I created

**Pull request closed without merging**